src/commcare_cloud/environment/secrets/backends/ansible_vault/main.py

```diff
@@ -12,53 +12,62 @@
 
 from commcare_cloud.environment.paths import ANSIBLE_DIR
 from commcare_cloud.environment.secrets.backends.abstract_backend import AbstractSecretsBackend
-from commcare_cloud.environment.secrets.secrets_schema import get_known_secret_specs
+from commcare_cloud.environment.secrets.secrets_schema import get_generated_variables
 
 
 class AnsibleVaultSecretsBackend(AbstractSecretsBackend):
-    def __init__(self, name, vault_file_path, record_to_datadog=False):
-        self.name = name
+    name = 'ansible-vault'
+
+    def __init__(self, env_name, vault_file_path, record_to_datadog=False, ask_vault_pass=True):
+        self.env_name = env_name
         self.vault_file_path = vault_file_path
         self.record_to_datadog = record_to_datadog
         self.should_send_vault_loaded_event = True
+        self.ask_vault_pass = ask_vault_pass
+
+    @classmethod
+    def from_environment(cls, environment):
+        try:
+            datadog_enabled = environment.public_vars.get('DATADOG_ENABLED')
+        except IOError:
+            # some test envs don't have public.yml
+            datadog_enabled = False
+
+        try:
+            commcare_cloud_use_vault = environment.public_vars.get('commcare_cloud_use_vault', True)
```

Comment:
with the ability to support different backends you could extract this out into a different class instead of customizing the vault backend:

```python
class EnvFileSecretsBackend(AbstractSecretsBackend)
    pass

class AnsibleVaultSecretsBackend(EnvFileSecretsBackend):
   pass
```

Just a thought

Reply:
That's not a bad idea. I'll hold off doing anything on that for now as I get this new backend working, but the next time we're changing something around that might be a good time to pull it out. At the very least it'd give us a place to write some doc strings about how you should never use it except for dummy environments like local or test environments